engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/AncientDataMotionStrategy.java

```diff
@@ -210,18 +213,19 @@ protected Answer copyObject(DataObject srcData, DataObject destData, Host destHo
      * @param dataTO Dest data store TO
      * @return dataTO including fullCloneFlag, if provided
      */
-    protected DataTO addFullCloneFlagOnVMwareDest(DataTO dataTO) {
+    protected DataTO addFullCloneAndDiskprovisiongStrictnessFlagOnVMwareDest(DataTO dataTO) {
         if (dataTO != null && dataTO.getHypervisorType().equals(Hypervisor.HypervisorType.VMware)){
             DataStoreTO dataStoreTO = dataTO.getDataStore();
             if (dataStoreTO != null && dataStoreTO instanceof PrimaryDataStoreTO){
                 PrimaryDataStoreTO primaryDataStoreTO = (PrimaryDataStoreTO) dataStoreTO;
-                Boolean value = CapacityManager.VmwareCreateCloneFull.valueIn(primaryDataStoreTO.getId());
-                primaryDataStoreTO.setFullCloneFlag(value);
+                primaryDataStoreTO.setFullCloneFlag(CapacityManager.VmwareCreateCloneFull.valueIn(primaryDataStoreTO.getId()));
+                primaryDataStoreTO.setDiskProvisioningStrictnessFlag(storageManager.DiskProvisioningStrictness.valueIn(primaryDataStoreTO.getId()));
             }
```

Comment:
DiskProvisioningStrictness setting is of zone scope, please use zone id here, instead of primaryDataStoreTO.getId()

Reply:
Done

Reply:
Done